libs/entitlement/asm-provider/src/main/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImpl.java

```diff
@@ -130,7 +135,13 @@ static MethodKey parseCheckerMethodSignature(String checkerMethodName, Type[] ch
             }
             targetParameterTypes = Arrays.stream(checkerMethodArgumentTypes).skip(2).map(Type::getInternalName).toList();
         }
-        boolean hasReceiver = (targetMethodIsStatic || targetMethodIsCtor) == false;
-        return new MethodKey(targetClassName, targetMethodName, targetParameterTypes);
+        return new MethodKey(checkerMethod.targetClassName(), checkerMethod.targetMethodName(), targetParameterTypes);
     }
+
+    private record ParsedCheckerMethod(
```

Comment:
My intention was to use this to implement [this TODO](https://github.com/elastic/elasticsearch/pull/119592/files#diff-00f96c1a4703ca4950b5e506d8fb5fb81758ede1c8da0161cfd39ae3d336e2b5R175). However, `InstrumentationServiceImpl` is not available at that point in the code, so we'd need to move this to a common location.

Deferring that for a subsequent PR seemed like a good idea, but if the team wants me to do it here in this PR, I can.